3-mcstas/mcstas-powder-diffraction.ipynb

```diff
@@ -230,34 +347,116 @@
+    "Now the simulation can be executed with the `backengine` method. This method returns a list of data object. Store this returned data in a python variable named `data`."
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "id": "35dcf5e1-0035-4651-a937-76d300db3805",
-   "metadata": {},
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "solution",
+     "hide-cell"
+    ]
+   },
    "outputs": [],
    "source": [
     "data = instrument.backengine()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "ccea6b4f-925b-44d4-8df5-52e712366e77",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "### Visualize the data\n",
+    "The data objects in the returned list can be plotted with the McStasScript function `make_sub_plot`.\n",
+    "\n",
+    "#### Plot overview\n",
+    "The function should plot three graphs:\n",
+    "##### PSD monitor\n",
+    "Shows spatial distribution of intensity at the sample position.\n",
+    "##### Time of flight monitor\n",
+    "Shows intensity as a function of neutron arrival time at the sample position.\n",
+    "##### Wavelength monitor\n",
+    "Shows intensity as a function of neutron wavelength at the sample position. Since a tiny wavelength interval was used, this simulation is almost for a constant wavelength.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2a1059e3-dde7-42d3-b2dd-cb10e2ce4d89",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "remove-cell",
+     "dmsc-school-keep"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# %matplotlib widget # Uncomment this to enable widget mode, this helps read values of graphs"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "id": "294ab064-2c47-467c-83fb-634b1930c533",
-   "metadata": {},
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "solution",
+     "hide-cell"
+    ]
+   },
    "outputs": [],
    "source": [
-    "%matplotlib widget\n",
-    "\n",
     "ms.make_sub_plot(data)"
    ]
   },
   {
    "cell_type": "markdown",
    "id": "f57ff9ba-05b6-4a2f-8e26-4dff1ade6b95",
-   "metadata": {},
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
    "source": [
-    "### Time resolution\n",
+    "### Time / wavelength resolution\n",
+    "In order to analyze data from this instrument, the neutron energy is needed and on a pulsed source this can be calculated from the arrival time of the neutrons and the known distance between the source and sample. Since the simulation was performed with a constant wavelength, the width of the time distribution can be used to deduce the uncertainty in the recorded energy.\n",
```

Comment:
```suggestion
    "In order to analyze data from this instrument, the neutron energy is needed. On a pulsed source, this can be calculated from the arrival time of the neutrons and the known distance between the source and sample. Since the simulation was performed with a constant wavelength, the width of the time distribution can be used to deduce the uncertainty in the computed energy.\n",
```